nixos/modules/services/web-apps/engelsystem.nix

```diff
@@ -0,0 +1,159 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  inherit (lib) mkDefault mkEnableOption mkIf mkOption types literalExample;
+
+  cfg = config.services.engelsystem;
+  fpm = config.services.phpfpm.pools.engelsystem;
+
+  genConfigScript = pkgs.writeScript "engelsystem-gen-config.sh" (utils.genJqSecretsReplacementSnippet cfg.config "config.json");
+in {
+  options = {
+    services.engelsystem = {
+      enable = mkEnableOption "Online tool for coordinating helpers and shifts on large events";
+
+      domain = mkOption {
+        type = types.str;
+        example = "engelsystem.example.com";
+        description = "domain to serve on";
+      };
+
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.engelsystem";
+        description = "Engelsystem package used for the service";
+        default = pkgs.engelsystem;
+      };
+
+      createDatabase = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to create a local database automatically.
+          This will overrid every database setting in <option>services.engelsystem.config</option>
+        '';
+      };
+    };
+
+    services.engelsystem.config = mkOption {
+      type = types.attrs;
+      default = {
+        database = {
+          unix_socket = "/var/run/mysqld/mysqld.sock";
+          host = "localhost";
+          database = "engelsystem";
+          username = "engelsystem";
+          password = "";
```

Comment:
I don't have a clue about php or the mysql adapter used here. I belive username is needed, but I don't know

Reply:
Yes, I think you're right about that.

If you rebase on master, you should now be able to get rid of the `unix_socket` option, since it defaults to the right one.

Reply:
On which master? not realy sure what you mean

Reply:
On `NixOS/nixpkgs`'s `master`-branch. If you don't have the `NixOS/nixpkgs` repo added as a remote, you can add it and rebase on `master` like this:
```sh
git remote add nixpkgs https://github.com/NixOS/nixpkgs.git
git pull --rebase nixpkgs master
```

Reply:
I'm not sure if I understand, what has nixpkgs to do with the options of engelsystem? how does it set it as a default?

Reply:
It's detailed in #84993, but essentially the `unix_socket` option is an argument to the `pdo_mysql` extension which should now default to the correct path, since it's specified when building the extension.

Reply:
ok, will test and rebase as soon as I have the time for it

Reply:
Rebased it, and it is working on my server